core/node/events/events.go

```diff
@@ -70,6 +70,33 @@ func MakeStreamEvent(
 	return event, nil
 }
 
+func MakeStreamEventWithTags(
+	wallet *crypto.Wallet,
+	payload IsStreamEvent_Payload,
+	prevMiniblockHash []byte,
+	Tags *Tags,
+) (*StreamEvent, error) {
+	salt := make([]byte, 32)
+	_, err := rand.Read(salt)
+	if err != nil {
+		return nil, AsRiverError(err, Err_INTERNAL).
+			Message("Failed to create random salt").
+			Func("MakeStreamEvent")
+	}
+	epochMillis := time.Now().UnixNano() / int64(time.Millisecond)
+
+	event := &StreamEvent{
+		CreatorAddress:    wallet.Address.Bytes(),
+		Salt:              salt,
+		PrevMiniblockHash: prevMiniblockHash,
```

Comment:
```suggestion
		PrevMiniblockHash: prevMiniblock.Hash[:],
		PrevMiniblockNum:  prevMiniblock.Num,
```
to match prev comment